src/ui/popup/manage-settings-page/import-settings.tsx

```diff
@@ -0,0 +1,62 @@
+import {m} from 'malevic';
+import {ViewProps} from '../types';
+import ControlGroup from '../control-group';
+import {UserSettings} from '../../../definitions';
+import {Button} from '../../controls';
+import {openFile} from '../../utils';
+import {DEFAULT_SETTINGS} from '../../../defaults';
+import {forEach} from '../../../utils/array';
+
+export default function ImportButton(props: ViewProps) {
+
+    function getValidatedObject(source: any, compare: any) {
+        const result = {};
+        forEach(Object.keys(source), (key) => {
+            const value: string = source[key];
+            const array1: boolean = Array.isArray(value)
+            const array2: boolean = Array.isArray(compare[key])
+            if (value == null || compare[key] == null) {
+                return;
```

Comment:
So, `array1` and `array2` become useless here, so move this condition to the beginning.